addons/PlayerMapMarkers/scripts/Game/FPM/Map/Markers/SCR_MapMarkerManagerComponent.c

```diff
@@ -2,21 +2,71 @@
 modded class SCR_MapMarkerManagerComponent : SCR_BaseGameModeComponent
 {
 	//------------------------------------------------------------------------------------------------
-	//! Copy of vanilla SetStreamRulesForPlayer, but added exception handling when marker is null
+	//! Similar to vanilla SetStreamRulesForPlayer, but added exception handling when marker is null and more advanced logic to determine whether to stream a marker
 	override void SetStreamRulesForPlayer(int playerID)
-	{			
+	{
 		foreach (SCR_MapMarkerEntity dynamicMarker : m_aDynamicMarkers)
 		{
 			// FPM: Added exception handling
 			if (!dynamicMarker)
 				continue;
 			
-			Faction markerFaction = dynamicMarker.GetFaction();
+			// Cast to FPM so we can get the player this marker is associated with
+			FPM_MapMarkerPlayer fpmMarker = FPM_MapMarkerPlayer.Cast(dynamicMarker);
+			if (!fpmMarker)
+				continue;
```

Comment:
This is no good. You basically break any other type of dynamic markers that way.

Reply:
Addressed in fb90c8a